code/modules/organs/organ.dm

```diff
@@ -145,22 +145,16 @@
 		handle_germ_effects()
 
 	if(owner && length(ailments))
-		for(var/datum/ailment/ailment in ailments)
-			if(!ailment.treated_by_reagent_type)
-				continue
-			var/treated
-			var/datum/reagents/bloodstr_reagents = owner.get_injected_reagents()
-			if(bloodstr_reagents)
-				if(REAGENT_VOLUME(bloodstr_reagents, ailment.treated_by_reagent_type) >= ailment.treated_by_reagent_dosage)
-					treated = bloodstr_reagents
-				else if(REAGENT_VOLUME(owner.reagents, ailment.treated_by_reagent_type) >= ailment.treated_by_reagent_dosage)
-					treated = owner.reagents
-				else
-					var/datum/reagents/ingested = owner.get_ingested_reagents()
-					if(ingested && REAGENT_VOLUME(ingested, ailment.treated_by_reagent_type) >= ailment.treated_by_reagent_dosage)
-						treated = ingested
-			if(treated)
-				ailment.was_treated_by_medication(treated)
+		ailment_loop:
```

Comment:
As discussed I think this should be split out into proc calls instead of using this structure. We generally do not permit goto to be used and this is functionally a goto.